clients/cli/src/account.rs

```diff
@@ -0,0 +1,166 @@
+use std::io::Write;
+
+use lb::Core;
+
+use crate::input;
+use crate::CliError;
+
+#[derive(clap::Subcommand, Debug)]
+pub enum AccountCmd {
+    /// create a new lockbook account
+    New {
+        /// your desired username (will prompt if not provided)
+        username: Option<String>,
+        /// the server url to register with (will default first to API_URL then to the lockbook
+        /// server)
+        api_url: Option<String>,
+    },
+    /// import an existing account by piping in the account string
+    Import,
+    /// reveal your account's private key
+    Export,
+    /// start a monthly subscription for massively increased storage
+    Subscribe,
+    /// cancel an existing subscription
+    Unsubscribe,
+    /// show your account status
+    Status,
+}
+
+pub fn account(core: &Core, cmd: AccountCmd) -> Result<(), CliError> {
+    match cmd {
+        AccountCmd::New { username, api_url } => new_acct(core, username, api_url),
+        AccountCmd::Import => import_acct(core),
+        AccountCmd::Export => export_acct(core),
+        AccountCmd::Subscribe => subscribe(core),
+        AccountCmd::Unsubscribe => unsubscribe(core),
+        AccountCmd::Status => status(core),
+    }
+}
+
+fn new_acct(
+    core: &Core, maybe_username: Option<String>, maybe_api_url: Option<String>,
+) -> Result<(), CliError> {
+    print!("please enter a username: ");
+    std::io::stdout().flush()?;
+
+    let username = match maybe_username {
+        Some(uname) => uname,
+        None => {
+            let mut uname = String::new();
+            std::io::stdin()
+                .read_line(&mut uname)
+                .expect("failed to read from stdin");
+            uname.retain(|c| c != '\n' && c != '\r');
```

Comment:
don't feel strongly, but maybe we want to chuck this business in a fn

Reply:
Which business, just reading the username from stdin?

Reply:
yeah looks like we do have that in the main.rs

Reply:
Ah, good call

Reply:
https://github.com/lockbook/lockbook/pull/1561/commits/7539bfd9738125308c984645cf3ff94ddc429c8c
(also fixed a future bug lol the prompt was there before I even checked for the optional username arg)